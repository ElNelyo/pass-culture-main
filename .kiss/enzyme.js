// ./node_modules/.bin/jest --env=jsdom ./path/to/file.spec.js --watch
import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // given
      const props = { errors: {} }
      // when
      const wrapper = shallow(<MyComponent {...props} />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
