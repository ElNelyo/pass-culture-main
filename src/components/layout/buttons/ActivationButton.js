/* eslint
  react/jsx-one-expression-per-line: 0 */
import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { noop } from '../../../utils/functionnals'

const ActivationButton = ({ children, className, onClickHandler, style }) => (
  <Link
    style={{ ...style }}
    to="/activation/events"
    onClick={onClickHandler}
    className={`activation-button ${className}`}
  >
    {children || <span className="is-block">Activer</span>}
  </Link>
)

ActivationButton.defaultProps = {
  children: null,
  className: '',
  onClickHandler: noop,
  style: {},
}

ActivationButton.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  onClickHandler: PropTypes.func,
  style: PropTypes.object,
}

export default ActivationButton
