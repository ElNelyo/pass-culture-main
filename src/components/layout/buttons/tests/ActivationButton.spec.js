/* eslint
  react/jsx-one-expression-per-line: 0 */
// jest --env=jsdom ./src/components/layout/buttons/tests/ActivationButton --watch
import React from 'react'
import { shallow } from 'enzyme'
import ActivationButton from '../ActivationButton'

describe('src | components | layout |Â buttons | ActivationButton', () => {
  describe('snapshot', () => {
    it('whithout props', () => {
      // when
      const wrapper = shallow(<ActivationButton />)
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
    it('whith props', () => {
      // given
      const props = {
        className: 'pc-theme-gradient px18 pb18 pt16 fs20',
        onClickHandler: jest.fn(),
        style: { borderRadius: 4 },
      }
      // when
      const wrapper = shallow(
        <ActivationButton {...props}>
          <span>Activez votre pass Culture</span>
        </ActivationButton>
      )
      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
