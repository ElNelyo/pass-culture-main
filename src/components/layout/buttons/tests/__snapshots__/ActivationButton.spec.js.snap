// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | layout | buttons | ActivationButton snapshot whith props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivationButton
    className="pc-theme-gradient px18 pb18 pt16 fs20"
    onClickHandler={[MockFunction]}
    style={
      Object {
        "borderRadius": 4,
      }
    }
  >
    <span>
      Activez votre pass Culture
    </span>
  </ActivationButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <span>
        Activez votre pass Culture
      </span>,
      "className": "activation-button pc-theme-gradient px18 pb18 pt16 fs20",
      "onClick": [MockFunction],
      "replace": false,
      "style": Object {
        "borderRadius": 4,
      },
      "to": "/activation/events",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Activez votre pass Culture",
      },
      "ref": null,
      "rendered": "Activez votre pass Culture",
      "type": "span",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <span>
          Activez votre pass Culture
        </span>,
        "className": "activation-button pc-theme-gradient px18 pb18 pt16 fs20",
        "onClick": [MockFunction],
        "replace": false,
        "style": Object {
          "borderRadius": 4,
        },
        "to": "/activation/events",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Activez votre pass Culture",
        },
        "ref": null,
        "rendered": "Activez votre pass Culture",
        "type": "span",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`src | components | layout | buttons | ActivationButton snapshot whithout props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActivationButton
    className=""
    onClickHandler={[Function]}
    style={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <span
        className="is-block"
      >
        Activer
      </span>,
      "className": "activation-button ",
      "onClick": [Function],
      "replace": false,
      "style": Object {},
      "to": "/activation/events",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Activer",
        "className": "is-block",
      },
      "ref": null,
      "rendered": "Activer",
      "type": "span",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <span
          className="is-block"
        >
          Activer
        </span>,
        "className": "activation-button ",
        "onClick": [Function],
        "replace": false,
        "style": Object {},
        "to": "/activation/events",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Activer",
          "className": "is-block",
        },
        "ref": null,
        "rendered": "Activer",
        "type": "span",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
